esphome/components/sensor/__init__.py

```diff
@@ -819,6 +822,23 @@ async def round_multiple_filter_to_code(config, filter_id):
     )
 
 
+@FILTER_REGISTRY.register(
+    "round_to_sigfig",
+    RoundSigfigFilter,
+    cv.maybe_simple_value(
+        {
+            cv.Required(CONF_DIGITS): cv.positive_not_null_int,
+        },
+        key=CONF_DIGITS,
+    ),
+)
+async def round_sigfig_filter_to_code(config, filter_id):
+    return cg.new_Pvariable(
+        filter_id,
+        config[CONF_DIGITS],
+    )
```

Comment:
Not sure this is the best name for the filter - "fig" doesn't gel with the "digits" used in the config, and generally speaking abbreviating words in config options is not ideal. Maybe "maximum_significant_digits"? It's wordy, but clear, and eliminates the need to have "digits" as a sub-key.


```suggestion
    "maximum_significant_digits",
    RoundSigfigFilter,
    cv.positive_not_null_int,
)
async def round_sigfig_filter_to_code(config, filter_id):
    return cg.new_Pvariable(
        filter_id,
        config,
    )
```